apps/presmooth.cpp

```diff
@@ -64,7 +64,7 @@ int main(int argc, char **argv) // TODO a quoi ça sert ?
 
     SourceMesh.load(argv[1]);
 
-    SparseMatrix SmoothMatrix = SourceMesh.gradient();
+    SparseMatrix SmoothMatrix = SourceMesh.gradient_norm2();
```

Comment:
Signature from `mesh3.h` gradient and `mesh.h` `gradient_norm2` is not the same. 

See when `gradient` changed to `gradient_norm` before becoming `gradient_norm2`:
```sh
commit 21c7424879629a07a5b86c9dac43509571c62710
Author: Emmanuel Olivi <emmanuel.olivi@gmail.com>
Date:   Wed Dec 18 18:06:01 2013 +0100

    Remove a lot of annoying warnings and add a gradient norm function.

diff --git a/src/mesh.h b/src/mesh.h
--- a/src/mesh.h
+++ b/src/mesh.h
@@ -172,3 +174,3 @@
         /// \brief Compute the surfacic gradient
-        void gradient(SparseMatrix &A) const;
+        void gradient_norm(SymMatrix &A, double coeff) const;
```